GMS-78 filtorwanie pól

PKopel opened the pull request:
### Zmiany
* specjalne serializator do klas `User` 
* specjalny agregator i serializator dla `Course` uzupełniający dane nauczycieli
* usunięcie pola `Course.isEnrolled`, wartości są generowane przez serializator
* inlineowanie pomocniczych funkcji

PKopel committed:
filtrowanie danych studentów i zapisywanie wielu ocen na raz

PKopel committed:
GMS-78 filtrowanie niepotrzebnych pól
specjalne serializatory do klas `User` i `Course`
usunięcie pola `Course.isEnrolled`, wartości są generowane przez serializator
inlineowanie pomocniczych funkcji

PKopel committed:
GMS-78 filtrowanie niepotrzebnych pól
specjalny agregator i serializator dla `Course` uzupełniający dane nauczycieli

PKopel committed:
Merge branch 'master' into test-db

pawel00100 commented on src/main/kotlin/com/gumi/moodle/model/ModelUtils.kt:
```diff
@@ -0,0 +1,29 @@
+object UserSerializer : JsonTransformingSerializer<User>(User.serializer()) {
+    override fun transformSerialize(element: JsonElement): JsonElement =
+        JsonObject(element.jsonObject.filterKeys { it != "password" && it != "salt" })
+}
```
Również raczej chcemy usuwać notyfikacje, w UserDAO są usuwane, poza pobieraniem dla enpointu od notyfikacji.

pawel00100 commented on src/main/kotlin/com/gumi/moodle/Utils.kt:
```diff
@@ -4,6 +4,7 @@ import com.gumi.moodle.dao.CourseDAO
 import io.ktor.application.*
 import io.ktor.auth.*
+import kotlinx.serialization.json.Json
 import org.koin.dsl.module
```
Zbędny nowy import (ogólnie w oknie commitu warto patrzeć na wszystkie zmienione pliki czy zmiany w nich mają sens)

pawel00100 commented on src/main/kotlin/com/gumi/moodle/rest_controllers/CourseController.kt:
```diff
@@ -85,17 +88,18 @@ fun Application.courseRoutes() {
 
-                            call.respond(course)
+                            println(course.teacherNames)
+                            if (isStudent) call.respond(CourseSerializer(userID), course)
```
print do usunięcia

PKopel committed:
GMS-78 filtrowanie niepotrzebnych pól
usuwanie notyfikacji z wysyłanych obiektów
wyczyszczony kod
stałe zamiast zahardcodowanych stringów
funkcja `getGrade` przeniesiona do `CourseDAO`
